lib/fx/adapters/postgres/functions.rb

```diff
@@ -19,7 +19,17 @@ class Functions
               ON pd.objid = pp.oid AND pd.deptype = 'e'
           LEFT JOIN pg_aggregate pa
               ON pa.aggfnoid = pp.oid
-          WHERE pn.nspname = 'public' AND pd.objid IS NULL
+          WHERE pn.nspname
+              IN (
+                  SELECT replace(
+                      unnest(string_to_array(setting, ', ')),
+                      '"$user"',
```

Comment:
I'm not seeing that Rails goes as far as resolving `"$user"` to whatever the current postgres username is when constructing its `--schema=` arguments for `structure.sql`. So this PR maybe goes a little beyond Rails in that respect, and I could buy unwinding this particular case and not supporting functions resolved by this special dynamic schema flag.

Reply:
I wonder if we can rewrite this SQL, and if this is time time to do it in a different way. This SQL was written pretty haphazardly by myself without knowing much how I would access these functions. I got it to work and moved on...

Now - I scoured through Scenic which served as a lot of inspiration for fx, and specifically how they resolve their views and filter them out:

https://github.com/scenic-views/scenic/blob/main/lib/scenic/adapters/postgres/views.rb

They don't access anything relative to the owner of the function, and I wonder if we can do something similar here.

Could you look into that and see what we can find to improve this?

They also differentiate defining views as `namespace.my_view` when namespace is no the default `public` - that could also be a thing fx could adapt.